src/main/resources/data/computercraft/lua/rom/programs/poster.lua

```diff
@@ -27,68 +37,137 @@ if args[1] == "stop" then
   return
 end
 
+local status = printer.status()
+if status == "busy" then
+  error("Printer is busy", 0)
+end
+
+-- Convert string to number and validate it.
+local function getNum(str)
+  local num = tonumber(str)
+  if not num or math.floor(num) < 1 then
+    return
+  end
+  return math.floor(num)
+end
+
+-- Get count.
 local count = 1
-if #args == 2 then
-  count = tonumber(args[2])
-  if not count or math.floor(count) < 1 then
+if #args > 1 then
+  count = getNum(args[2])
+  if not count then
     printUsage()
     return
   end
-  count = math.floor(count)
 end
 
--- Load the 2dj file
+-- Get index, if given.
+local index = { all = true, from = nil, to = nil }
+if #args >= 3 then
+  local num1, num2 = getNum(args[3]), getNum(args[4])
+  if not num1 or not num2 then
+    printUsage()
+    return
+  end
+  
+  index.from = math.min(num1, num2)
+  index.to = math.max(num1, num2)
+  
+  -- Index, not interval
+  if not index.to then
+    index.to = index.from
+  end
+  
+  if index.from then
+    index.all = false
+  end
+end
+
 local filename = args[1]
+-- Loads from local file or Url
+local function loadPoster(name)
+  local stream
 
-local function load2djFile(name)
-  local f
   if http and name:match("^https?://") then
     print("Downloading...")
-    f = http.get{ url = name, binary = true }
+    stream = http.get{ url = name, binary = true }
   else
+    print("Loading...")
     name = shell.resolve(name)
     if not fs.exists(name) then
-      -- Try to find a .2dj file if the user didn't specify the extension
```

Comment:
This should either be kept or configured as a `poster.default_extension` setting IMO. No point in removing it.

Reply:
I don't think it should be a setting, but I agree we should not remove this behavior. Sorry if I was unclear in my previous comment. The behavior should be as follows:

1. Attempt to resolve the name as is
2. If that fails, attempt to resolve **both** `.2dj` and `.2dja`
  a. If both exist, error and require the user to specify
  b. If only `.2dj` (no `.2dja`) exists, use that
  c. If only `.2dja` (no `.2dj`) exists, use that
  d. Obviously if neither exist, error

Reply:
Ah, yeah that makes more sense. My bad.